File: Day-1/bhaskar0120.pl
# Counting Calories

sub max{
  ($a, $b) = @_;
  if ($a >= $b){
    return $a;
  }
  else{
    return $b;
  }
}

$ans1 = 0;

open($f,"inp.txt") or die "File not found";
  until(eof $f){
    $line = readline $f or die "Couldn't readline";
    if($line == "\n"){
      $ans1 = max $ans1, $sum;
      $sum = 0;
    }
    else{
      chomp $line;
      $sum += $line;
    }
  }
close $f or warn "Unable to close file";

$ans1 = max $ans1, $sum;
print "Easy: $ans1\n";

@maximum_array = ();
$sum = 0;

open($f,"inp.txt") or die "File not found";
  until(eof $f){
    $line = readline $f or die "Couldn't readline";
    if($line == "\n"){
      push @maximum_array, $sum;
      $sum = 0;
    }
    else{
      chomp $line;
      $sum += $line;
    }
  }
close $f or warn "Unable to close file";

push @maximum_array, $sum;

@sorted_array = reverse sort {$a <=> $b} @maximum_array;

$ans2 = @sorted_array[0] + @sorted_array[1] + @sorted_array[2];
print "Hard: $ans2\n";

File: Day-6/bhaskar0120.lua

-- Get input 
f = io.open('inp.txt', 'r')
s = f:read('line')
f:close()

array = {}
function convertToIterable()
  local count = 0
  for i in s:gmatch('.') do
    array[count] = i
    count = count + 1
  end
  len = count
end

function Solution(r)
  local found = {}
  local count = 0 --Count of unique character in sliding window
  for i=0,len-1 do
    if found[array[i]] == nil or found[array[i]] == 0 then 
      found[array[i]] = 1 
      count = count + 1
    else
      found[array[i]] = found[array[i]] + 1
    end

    if i > (r-1) then 
      if found[array[i-r]] == 1 then
        count = count - 1
      end
      found[array[i-r]] = found[array[i-r]] - 1
    end
    if count == r then 
      return i+1
    end
  end
end


convertToIterable()

print("Easy: "..Solution(4))
print("Hard: "..Solution(14))
